src/backend/gl/src/device.rs

```diff
@@ -366,22 +522,44 @@ impl d::Device<B> for Device {
         unimplemented!()
     }
 
-    fn acquire_mapping_raw(&mut self, buf: &n::Buffer, read: Option<Range<u64>>)
+    fn acquire_mapping_raw(&mut self, buffer: &n::Buffer, read: Option<Range<u64>>)
         -> Result<*mut u8, mapping::Error>
     {
-        unimplemented!()
+        let gl = &self.share.context;
+        let access = match read {
+            Some(_) => gl::READ_ONLY,
+            None => gl::WRITE_ONLY,
+        };
+
+        let data = unsafe {
+            gl.BindBuffer(buffer.target, buffer.raw);
+            let ptr = gl.MapBuffer(buffer.target, access);
```

Comment:
Could we add a safety check at the beginning? Not all GL versions support glMapBuffer, none for GL ES.